rfqmsg/buy_request.go

```diff
@@ -286,28 +286,61 @@ func NewBuyRequest(peer route.Vertex, assetID *asset.ID,
 }
 
 // NewBuyRequestMsgFromWire instantiates a new instance from a wire message.
-func NewBuyRequestMsgFromWire(wireMsg WireMessage) (*BuyRequest, error) {
+func NewBuyRequestMsgFromWire(wireMsg WireMessage,
+	msgData requestWireMsgData) (*BuyRequest, error) {
+
 	// Ensure that the message type is a quote request message.
-	if wireMsg.MsgType != MsgTypeBuyRequest {
+	if wireMsg.MsgType != MsgTypeRequest {
 		return nil, fmt.Errorf("unable to create a buy request "+
 			"message from wire message of type %d", wireMsg.MsgType)
 	}
 
-	var msgData buyRequestMsgData
-	err := msgData.Decode(bytes.NewBuffer(wireMsg.Data))
-	if err != nil {
-		return nil, fmt.Errorf("unable to decode incoming buy "+
-			"request message data: %w", err)
+	var assetID *asset.ID
+	msgData.InAssetID.WhenSome(
+		func(inAssetID tlv.RecordT[tlv.TlvType5, asset.ID]) {
+			assetID = &inAssetID.Val
+		},
+	)
+
+	var assetGroupKey *btcec.PublicKey
+	msgData.InAssetGroupKey.WhenSome(
+		// nolint: lll
+		func(inAssetGroupKey tlv.RecordT[tlv.TlvType6, *btcec.PublicKey]) {
```

Comment:
nit: feel free to use less descriptive parameter names for such closures if it helps to avoid running into the length limit. For example just `key` instead of `inAssetGroupKey`, since from the outer context it's clear what the variable means.
Same further below (`t` instead of `suggestedRateTick`).